roboteam_ai/src/world/Robot.cpp

```diff
@@ -12,24 +12,26 @@ namespace rtt {
 namespace ai {
 namespace world {
 
-std::map<int, unsigned char> Robot::genevaState;
-
-Robot::Robot(const roboteam_msgs::WorldRobot &copy, Team team, unsigned long worldNumber)
-        : distanceToBall(-1.0), iHaveBall(false), id(copy.id), angle(copy.angle),
-          pos(copy.pos), vel(copy.vel), angularVelocity(copy.w), team(team), lastUpdatedWorldNumber(worldNumber) {
-
-    if (id != -1 && genevaState.find(id) == genevaState.end()) {
-        genevaState[id] = 3;
+Robot::Robot(const roboteam_msgs::WorldRobot &copy, Team team,
+        unsigned char genevaState, unsigned char dribblerState, unsigned long worldNumber)
+        :distanceToBall(- 1.0), iHaveBall(false), lastUpdatedWorldNumber(worldNumber), genevaState(genevaState),
+         dribblerState(dribblerState),
+         id(copy.id), angle(copy.angle), pos(copy.pos), vel(copy.vel), angularVelocity(copy.w), team(team) {
+
+    if (id > - 1 && id < 16) {
+        workingGeneva = Constants::ROBOT_HAS_WORKING_GENEVA(id);
+        workingDribbler = Constants::ROBOT_HAS_WORKING_DRIBBLER(id);
     }
-
-    if (id != -1) {
-        hasWorkingGeneva = Constants::ROBOT_HAS_WORKING_GENEVA(id);
+    else {
+        workingGeneva = false;
+        workingDribbler = false;
     }
 }
 
 Robot::Robot()
-        : distanceToBall(-1.0), iHaveBall(false), id(-1), angle(-1.0),
-          angularVelocity(-1.0), team(invalid), lastUpdatedWorldNumber(0) {
+        :distanceToBall(- 1.0), iHaveBall(false), lastUpdatedWorldNumber(0), genevaState(0), workingGeneva(false),
```

Comment:
would prefer if you initialize angle and angularvelocity with 0.0

Reply:
does not matter, is - and should not be - used in any other way than creating an empty robot.